src/nvim/ex_cmds.c

```diff
@@ -283,8 +283,8 @@ static int sort_abort;                  /* flag to indicate if sorting has been
 /* Struct to store info to be sorted. */
 typedef struct {
   linenr_T lnum;                        /* line number */
-  long start_col_nr;                    /* starting column number or number */
-  long end_col_nr;                      /* ending column number */
+  NumOpt start_col_nr;                  /* starting column number or number */
```

Comment:
I think it is better to leave this, and instead `vim_str2nr` to a temporary and then convert. (If anything,  these should be changed to `colnr_T` but that's a change for another PR)